README.md
# Table of Contents
- [Introduction](#introduction)
    - [Version](#version)
    - [Changelog](Changelog.md)
- [Hardware Requirements](#hardware-requirements)
    - [CPU](#cpu)
    - [Memory](#memory)
    - [Storage](#storage)
- [Contributing](#contributing)
- [Issues](#issues)
- [Installation](#installation)
- [Quick Start](#quick-start)
- [Configuration](#configuration)
  - [ACCESS](#access)
  - [QUEUE](#queue)
  - [TOPIC](#topic)
  - [Data Store](#data-store)
  - [BROKER](#broker)
  - [Disk usage](#disk-usage)
  - [JMX](#JMX)
  - [Avaible Configuration Parameters](#avaible-configuration-parameters)
  - [Advance configuration](#advance-configuration)
- [References](#references)

# Introduction

Dockerfile to build a ActiveMQ container image.

## Version

Current Version: **5.14.2**

# Hardware Requirements

## CPU

- No stats avaible to say the number of core in function of messages

## Memory

- 512MB is too little memory, i think is to use ActiveMQ on test environment
- **1GB** is the **standard** memory size

You can set the memory that you need :

```bash
docker run --name='activemq' -it --rm \
	-e 'ACTIVEMQ_MIN_MEMORY=512' \
	-e 'ACTIVEMQ_MAX_MEMORY=2048'\
        -P
	webcenter/activemq:latest
```
This sample lauch ActiveMQ in docker with 512 MB of memory, and then ACtiveMQ can take 2048 MB of max memory

## Storage

The necessary hard drive space depends if you use persistant message or not and the type of appender. Normaly, no need space for ActiveMQ because the most data are contains directly on memory.
I think it depends on how you use ActiveMQ ;)

# Contributing

If you find this image useful here's how you can help:

- Send a Pull Request with your awesome new features and bug fixes
- Help new users with [Issues](https://github.com/disaster37/activemq/issues) they may encounter
- Send me a tip via [Bitcoin](https://www.coinbase.com/disaster37) or using [Gratipay](https://gratipay.com/disaster37/)

# Issues

Docker is a relatively new project and is active being developed and tested by a thriving community of developers and testers and every release of docker features many enhancements and bugfixes.

Given the nature of the development and release cycle it is very important that you have the latest version of docker installed because any issue that you encounter might have already been fixed with a newer docker release.

For ubuntu users I suggest [installing docker](https://docs.docker.com/installation/ubuntulinux/) using docker's own package repository since the version of docker packaged in the ubuntu repositories are a little dated.

Here is the shortform of the installation of an updated version of docker on ubuntu.

```bash
sudo apt-get purge docker.io
curl -s https://get.docker.io/ubuntu/ | sudo sh
sudo apt-get update
sudo apt-get install lxc-docker
```

Fedora and RHEL/CentOS users should try disabling selinux with `setenforce 0` and check if resolves the issue. If it does than there is not much that I can help you with. You can either stick with selinux disabled (not recommended by redhat) or switch to using ubuntu.

If using the latest docker version and/or disabling selinux does not fix the issue then please file a issue request on the [issues](https://github.com/disaster/activemq/issues) page.

In your issue report please make sure you provide the following information:

- The host distribution and release version.
- Output of the `docker version` command
- Output of the `docker info` command
- The `docker run` command you used to run the image (mask out the sensitive bits).

# Installation

Pull the image from the docker index. This is the recommended method of installation as it is easier to update image. These builds are performed by the **Docker Trusted Build** service.

```bash
docker pull webcenter/activemq:5.14.2
```

You can also pull the `latest` tag which is built from the repository *HEAD*

```bash
docker pull webcenter/activemq:latest
```

Alternately you can build the image locally.

```bash
git clone https://github.com/disaster37/activemq.git
cd activemq
docker build --tag="$USER/activemq" .
```

# Quick Start

You can launch the image using the docker command line :

- **For test purpose :**

```bash
docker run --name='activemq' -it --rm -P \
webcenter/activemq:latest
```
The account admin is "admin" and password is "admin". All settings is the default ActiveMQ's settings.

- **For production purpose :**

```bash
docker run --name='activemq' -d \
-e 'ACTIVEMQ_NAME=amqp-srv1' \
-e 'ACTIVEMQ_REMOVE_DEFAULT_ACCOUNT=true' \
-e 'ACTIVEMQ_ADMIN_LOGIN=admin' -e 'ACTIVEMQ_ADMIN_PASSWORD=your_password' \
-e 'ACTIVEMQ_WRITE_LOGIN=producer_login' -e 'ACTIVEMQ_WRITE_PASSWORD=producer_password' \
-e 'ACTIVEMQ_READ_LOGIN=consumer_login' -e 'ACTIVEMQ_READ_PASSWORD=consumer_password' \
-e 'ACTIVEMQ_JMX_LOGIN=jmx_login' -e 'ACTIVEMQ_JMX_PASSWORD=jmx_password' \
-e 'ACTIVEMQ_STATIC_TOPICS=topic1;topic2;topic3' \
-e 'ACTIVEMQ_STATIC_QUEUES=queue1;queue2;queue3' \
-e 'ACTIVEMQ_MIN_MEMORY=1024' -e  'ACTIVEMQ_MAX_MEMORY=4096' \
-e 'ACTIVEMQ_ENABLED_SCHEDULER=true' \
-v /data/activemq:/data/activemq \
-v /var/log/activemq:/var/log/activemq \
-p 8161:8161 \
-p 61616:61616 \
-p 61613:61613 \
webcenter/activemq:5.14.2
```


Or you can use [fig](http://www.fig.sh/). Assuming you have fig installed,

```bash
wget https://raw.githubusercontent.com/disaster37/activemq/master/fig.yml
fig up
```

# Configuration

## ACCESS
todo


## QUEUE
todo

## TOPIC
todo

## Data Store
todo

## BROKER
todo

## Disk usage
todo

## JMX
todo

## Avaible Configuration Parameters

*Please refer the docker run command options for the `--env-file` flag where you can specify all required environment variables in a single file. This will save you from writing a potentially long docker run command. Alternately you can use fig.*

Below is the complete list of available options that can be used to customize your activemq installation.

- **ACTIVEMQ_NAME**: The hostname of ActiveMQ server. Default to `localhost`
- **ACTIVEMQ_LOGLEVEL**: The log level. Default to `INFO`
- **ACTIVEMQ_PENDING_MESSAGE_LIMIT**: It is used to prevent slow topic consumers to block producers and affect other consumers by limiting the number of messages that are retained. Default to `1000`
- **ACTIVEMQ_STORAGE_USAGE**: The maximum amount of space storage the broker will use before disabling caching and/or slowing down producers. Default to `100 gb`
- **ACTIVEMQ_TEMP_USAGE**: The maximum amount of space temp the broker will use before disabling caching and/or slowing down producers. Default to `50 gb`
- **ACTIVEMQ_MAX_CONNECTION**: It's DOS protection. It limit concurrent connections. Default to `1000`
- **ACTIVEMQ_FRAME_SIZE**: It's DOS protection. It limit the frame size. Default to `104857600` (100MB)
- **ACTIVEMQ_ENABLED_SCHEDULER**: Permit to enabled scheduler in ActiveMQ. Default to `true`
- **ACTIVEMQ_ENABLED_AUTH**: Permit to enabled the authentification in queue and topic (no anonymous access). Default to `false`
- **ACTIVEMQ_MIN_MEMORY**: The init memory in MB that ActiveMQ take when start (it's like XMS). Default to `128` (128 MB)
- **ACTIVEMQ_MAX_MEMORY**: The max memory in MB that ActiveMQ can take (it's like XMX). Default to `1024` (1024 MB)

- **ACTIVEMQ_REMOVE_DEFAULT_ACCOUNT**: It's permit to remove all default login on ActiveMQ (Webconsole, broker and JMX). Default to `false`
- **ACTIVEMQ_ADMIN_LOGIN**: The login for admin account (broker and web console). Default to `admin`
- **ACTIVEMQ_ADMIN_PASSWORD**: The password for admin account. Default to `admin`
- **ACTIVEMQ_USER_LOGIN**: The login to access on web console with user role (no right on broker). Default to `user`
- **ACTIVEMQ_USER_PASSWORD**: The password for user account. Default to `user`
- **ACTIVEMQ_READ_LOGIN**: The login to access with read only role on all queues and topics.
- **ACTIVEMQ_READ_PASSWORD**: The password for read account.
- **ACTIVEMQ_WRITE_LOGIN**: The login to access with write role on all queues and topics.
- **ACTIVEMQ_WRITE_PASSWORD**: The password for write account.
- **ACTIVEMQ_OWNER_LOGIN**: The login to access with admin role on all queues and topics.
- **ACTIVEMQ_OWNER_PASSWORD**: The password for owner account.
- **ACTIVEMQ_JMX_LOGIN**: The login to access with read / write role on JMX. Default to `admin`
- **ACTIVEMQ_JMX_PASSWORD**: The password for JMX account. Default to `activemq`

- **ACTIVEMQ_STATIC_TOPICS**: The list of topics separated by comma witch is created when ActiveMQ start.
- **ACTIVEMQ_STATIC_QUEUES**: The list of queues separated by comma witch is created when ActiveMQ start.


## Advance configuration

For advance configuration, the best way is to read ActiveMQ documentation and created your own setting file like activemq.xml.
Next, you can mount it when you run this image or you can create your own image (base on this image) and include your specifics config file.

The home of ActiveMQ is in /opt/activemq, so if you want to override all the setting, you can launch docker with ` -v /your_path/conf:/opt/activemq/conf`


Changelog.md
# Changelog

**3.0.3
  - Don't use cron or logrotate
  - User custom log4j.properties
   - Logging rotation should be handled outside of this image

**v3.0.2
  - Upgrade to ActiveMQ 5.13.2
  - Add parameter to disable auth

**v3.0.0
  - Upgrade to Ubuntu 15.10 and on last OpenJDK-jre
  - Upgrade to Activemq 5.12.0

**v2.0.2
  - Fix bug when you use persistant container

**v2.0.1
  - Add feature to enabled sheduler with parameter (#8)

**v2.0.0
  - Replace init bash script with python script. Integrate many unit test and patch on the flow the standard setting files

**v1.0.2
 - Change bash script to python script and introduce unit test to simplify the upgrade
 - Fix the issue #4 (Cannot mount a custom activemq.xml

**v1.0.1
- Add environment variabale to set MIN and MAX memory for ActiveMQ (issue #1)

**v1.0.0**
- upgraded to ActiveMQ 5.10.0


test/test.js
var Stomp = require('stomp-client');
var destination = '/topic/topic1';
var client = new Stomp('127.0.0.1', 61613);

client.connect(function(sessionId) {
    client.subscribe(destination, function(body, headers) {
      console.log('This is the body of a message on the subscribed queue:', body);
    });

    client.publish(destination, 'Oh herrow');
});


assets/entrypoint/entrypoint/Init.py
#!/usr/bin/python

import fileinput
import sys
import os
import shutil
import re


ACTIVEMQ_HOME = "/opt/activemq"
ACTIVEMQ_CONF = ACTIVEMQ_HOME + '/conf.tmp'


class Init():


    def replace_all(self, file, searchRegex, replaceExp):
      """ Replace String in file with regex
      :param file: The file name where you should to modify the string
      :param searchRegex: The pattern witch must match to replace the string
      :param replaceExp: The string replacement
      :return:
      """

      regex = re.compile(searchRegex, re.IGNORECASE)

      f = open(file,'r')
      out = f.readlines()
      f.close()

      f = open(file,'w')

      for line in out:
          if regex.search(line) is not None:
            line = regex.sub(replaceExp, line)

          f.write(line)

      f.close()


    def add_end_file(self, file, line):
        """ Add line at the end of file

        :param file: The file where you should to add line to the end
        :param line: The line to add in file
        :return:
        """
        with open(file, "a") as myFile:
            myFile.write("\n" + line + "\n")



    def do_setting_activemq_users(self, login, password):
        global ACTIVEMQ_HOME

        if login is None or login == "" :
            raise KeyError("You must set the login")
        if password is None or password == "":
            raise KeyError("You must set the password")

        self.add_end_file(ACTIVEMQ_CONF + "/users.properties", login + "=" + password)

    def do_remove_default_account(self):
        global ACTIVEMQ_HOME


        self.replace_all(ACTIVEMQ_CONF + "/users.properties","admin=admin", "")
        self.replace_all(ACTIVEMQ_CONF + "/jetty-realm.properties", "admin: admin, admin", "")
        self.replace_all(ACTIVEMQ_CONF + "/jetty-realm.properties", "user: user, user", "")
        self.replace_all(ACTIVEMQ_CONF + "/groups.properties", "admins=admin", "")
        self.replace_all(ACTIVEMQ_CONF + "/jmx.access", "admin readwrite", "")
        self.replace_all(ACTIVEMQ_CONF + "/jmx.password", "admin activemq", "")
        self.replace_all(ACTIVEMQ_CONF + "/credentials.properties", "activemq\.username=system", "")
        self.replace_all(ACTIVEMQ_CONF + "/credentials.properties", "activemq\.password=manager", "")
        self.replace_all(ACTIVEMQ_CONF + "/credentials.properties", "guest\.password=password", "")



    def do_setting_activemq_credential(self, user, password):
        global ACTIVEMQ_HOME

        if user is None or user == "" :
    	       raise KeyError("You must set the user")

        if password is None or password == "" :
    	       raise KeyError("You must set the password")

        self.add_end_file(ACTIVEMQ_CONF + "/credentials.properties", "activemq.username=" + user)
        self.add_end_file(ACTIVEMQ_CONF + "/credentials.properties", "activemq.password=" + password)



    def do_setting_activemq_groups(self, group, users):
        global ACTIVEMQ_HOME

        if group is None or group == "" :
            raise KeyError("You must set the group")

        if users is None:
            self.add_end_file(ACTIVEMQ_CONF + "/groups.properties", group + "=")
        else:
            self.add_end_file(ACTIVEMQ_CONF + "/groups.properties", group + "=" + users)


    def do_setting_activemq_web_access(self, role, user, password):
        global ACTIVEMQ_HOME

        if role is None or role == "":
            raise KeyError("You must set the role")

        if user is None or user == "":
            raise KeyError("You must set the username")

        if password is None or password == "":
            raise KeyError("You must set the password")


        self.add_end_file(ACTIVEMQ_CONF + "/jetty-realm.properties", user + ": " + password + ", " +role);


    def do_setting_activemq_jmx_access(self, role, user, password):
        global ACTIVEMQ_HOME

        if role is None or role == "":
            raise KeyError("You must set role")

        if user is None or user == "":
            raise KeyError("You must set user")

        if password is None or password == "":
            raise KeyError("You must set password")

        self.add_end_file(ACTIVEMQ_CONF + "/jmx.access", user + " " + role)
        self.add_end_file(ACTIVEMQ_CONF + "/jmx.password", user + " " + password)


    def do_setting_activemq_log4j(self, loglevel):
        global ACTIVEMQ_HOME

        if loglevel is None or loglevel == "":
            raise KeyError("You must set loglevel")

        self.replace_all(ACTIVEMQ_CONF + "/log4j.properties", "log4j\.rootLogger=[^,]+", "log4j.rootLogger=" + loglevel)
        self.replace_all(ACTIVEMQ_CONF + "/log4j.properties", "log4j\.logger\.org\.apache\.activemq\.audit=[^,]+", "log4j.logger.org.apache.activemq.audit=" + loglevel)


    def do_setting_activemq_main(self, name, messageLimit, storageUsage, tempUsage, maxConnection, frameSize, topics, queues, enabledScheduler, enabledAuth):

        if name is None or name == "":
            raise KeyError("You must set the name")

        if messageLimit is None or messageLimit < 0:
            raise KeyError("You must set the messageLimit")

        if storageUsage is None or storageUsage == "":
            raise KeyError("You must set the storageUsage")

        if tempUsage is None or tempUsage == "":
            raise KeyError("You must set the tempStorage")

        if maxConnection is None or maxConnection < 0:
            raise KeyError("You must set the maxConnection")

        if frameSize is None or frameSize < 0:
            raise KeyError("You must set the frameSize")

        self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", 'brokerName="[^"]*"', 'brokerName="' + name + '"')
        self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", '<constantPendingMessageLimitStrategy limit="\d+"/>', '<constantPendingMessageLimitStrategy limit="' + str(messageLimit) + '"/>')
        self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", '<storeUsage limit="[^"]+"/>', '<storeUsage limit="' + storageUsage + '"/>')
        self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", '<tempUsage limit="[^"]+"/>', '<tempUsage limit="' + tempUsage + '"/>')
        self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", '\?maximumConnections=1000', "?maximumConnections=" + str(maxConnection))
        self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", 'wireFormat\.maxFrameSize=104857600', "wireFormat.maxFrameSize=" + str(frameSize))

        # Look for enabled scheduler
        if enabledScheduler == "true" :
        	self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", '<broker', '<broker schedulerSupport="true"')

        # We inject the setting to manage right on topic and queue if needed
        if enabledAuth == "true":
            rightManagement = """
<plugins>
    <!--  use JAAS to authenticate using the login.config file on the classpath to configure JAAS -->
    <jaasAuthenticationPlugin configuration="activemq" />
    <authorizationPlugin>
        <map>
            <authorizationMap>
                <authorizationEntries>
                    <authorizationEntry queue=">" read="admins,reads,writes,owners" write="admins,writes,owners" admin="admins,owners" />
                    <authorizationEntry topic=">" read="admins,reads,writes,owners" write="admins,writes,owners" admin="admins,owners" />
                    <authorizationEntry topic="ActiveMQ.Advisory.>" read="admins,reads,writes,owners" write="admins,reads,writes,owners" admin="admins,reads,writes,owners"/>
                </authorizationEntries>

                <!-- let's assign roles to temporary destinations. comment this entry if we don't want any roles assigned to temp destinations  -->
                <tempDestinationAuthorizationEntry>
                    <tempDestinationAuthorizationEntry read="tempDestinationAdmins" write="tempDestinationAdmins" admin="tempDestinationAdmins"/>
                </tempDestinationAuthorizationEntry>
            </authorizationMap>
        </map>
    </authorizationPlugin>
</plugins>\n"""
            self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", '</broker>', rightManagement + '</broker>')



        if (topics is not None and topics != "") or (queues is not None and queues != ""):
            staticRoute = "<destinations>\n"
            if topics is not None and topics != "" :
                topicList = topics.split(';')
                for topic in topicList:
                    staticRoute += '<topic physicalName="' + topic + '" />' + "\n"

            if queues is not None and queues != "":
                queueList = queues.split(';')
                for queue in queueList:
                    staticRoute += '<queue physicalName="' + queue + '" />' + "\n"

            staticRoute += "</destinations>\n"

            self.replace_all(ACTIVEMQ_CONF + "/activemq.xml", '</broker>', staticRoute + '</broker>')

    def do_setting_activemq_wrapper(self, minMemoryInMB, maxMemoryInMb):

        if minMemoryInMB is None or minMemoryInMB < 0:
            raise KeyError("You must set the minMemory")

        if maxMemoryInMb is None or maxMemoryInMb < 0:
            raise KeyError("You must set the maxMemory")

        self.replace_all(ACTIVEMQ_HOME + "/bin/linux-x86-64/wrapper.conf", "#?wrapper\.java\.initmemory=\d+", 'wrapper.java.initmemory=' + str(minMemoryInMB))
        self.replace_all(ACTIVEMQ_HOME + "/bin/linux-x86-64/wrapper.conf", "#?wrapper\.java\.maxmemory=\d+", 'wrapper.java.maxmemory=' + str(maxMemoryInMb))


    def do_init_activemq(self):

        # We change the activemq launcher to start activemq with activmq user
        self.replace_all(ACTIVEMQ_HOME + "/bin/linux-x86-64/activemq", "#RUN_AS_USER=", "RUN_AS_USER=activemq")

        # We change some macro on wrapper.conf to move data
        self.replace_all(ACTIVEMQ_HOME + "/bin/linux-x86-64/wrapper.conf" ,"set\.default\.ACTIVEMQ_DATA=%ACTIVEMQ_BASE%\/data", "set.default.ACTIVEMQ_DATA=/data/activemq")

        # Fix bug #4 "Cannot mount a custom activemq.xml"
        self.replace_all(ACTIVEMQ_HOME + "/bin/linux-x86-64/wrapper.conf" ,"set\.default\.ACTIVEMQ_CONF=%ACTIVEMQ_BASE%/conf$", "set.default.ACTIVEMQ_CONF=%ACTIVEMQ_BASE%/conf.tmp")

        # We replace the log output
        self.replace_all(ACTIVEMQ_CONF + "/log4j.properties", "\$\{activemq\.base\}\/data\/", "/var/log/activemq/")
        self.replace_all(ACTIVEMQ_HOME + "/bin/linux-x86-64/wrapper.conf" ,"wrapper\.logfile=%ACTIVEMQ_DATA%\/wrapper\.log", "wrapper.logfile=/var/log/activemq/wrapper.log")


    def setting_all(self):
        # We look if we must remove default account
        if os.getenv('ACTIVEMQ_REMOVE_DEFAULT_ACCOUNT') == "true":
            self.do_remove_default_account()

        # We init some fix parameter
        self.do_init_activemq()

        # We setting the admin account
        if os.getenv('ACTIVEMQ_ADMIN_LOGIN') is not None and os.getenv('ACTIVEMQ_ADMIN_PASSWORD') is not None:
            self.do_setting_activemq_users(os.getenv('ACTIVEMQ_ADMIN_LOGIN'),
                                                 os.getenv('ACTIVEMQ_ADMIN_PASSWORD'))
            self.do_setting_activemq_web_access("admin", os.getenv('ACTIVEMQ_ADMIN_LOGIN'),
                                                      os.getenv('ACTIVEMQ_ADMIN_PASSWORD'))
            self.do_setting_activemq_groups("admins", os.getenv('ACTIVEMQ_ADMIN_LOGIN'))
            self.do_setting_activemq_credential(os.getenv('ACTIVEMQ_ADMIN_LOGIN'),
                                                      os.getenv('ACTIVEMQ_ADMIN_PASSWORD'))

        # We setting the user account
        if os.getenv('ACTIVEMQ_USER_LOGIN') is not None and os.getenv('ACTIVEMQ_USER_PASSWORD') is not None:
            self.do_setting_activemq_users(os.getenv('ACTIVEMQ_USER_LOGIN'), os.getenv('ACTIVEMQ_USER_PASSWORD'))
            self.do_setting_activemq_web_access("user", os.getenv('ACTIVEMQ_USER_LOGIN'),
                                                      os.getenv('ACTIVEMQ_USER_PASSWORD'))

        # We setting the owner account
        if os.getenv('ACTIVEMQ_OWNER_LOGIN') is not None and os.getenv('ACTIVEMQ_OWNER_PASSWORD') is not None:
            self.do_setting_activemq_users(os.getenv('ACTIVEMQ_OWNER_LOGIN'),
                                                 os.getenv('ACTIVEMQ_OWNER_PASSWORD'))
            self.do_setting_activemq_groups("owners", os.getenv('ACTIVEMQ_OWNER_LOGIN'))

        # We setting the writer account
        if os.getenv('ACTIVEMQ_WRITE_LOGIN') is not None and os.getenv('ACTIVEMQ_WRITE_PASSWORD') is not None:
            self.do_setting_activemq_users(os.getenv('ACTIVEMQ_WRITE_LOGIN'),
                                                 os.getenv('ACTIVEMQ_WRITE_PASSWORD'))
            self.do_setting_activemq_groups("writes", os.getenv('ACTIVEMQ_WRITE_LOGIN'))

        # We setting the reader account
        if os.getenv('ACTIVEMQ_READ_LOGIN') is not None and os.getenv('ACTIVEMQ_READ_PASSWORD') is not None:
            self.do_setting_activemq_users(os.getenv('ACTIVEMQ_READ_LOGIN'), os.getenv('ACTIVEMQ_READ_PASSWORD'))
            if os.getenv('ACTIVEMQ_USER_LOGIN') is not None and os.getenv('ACTIVEMQ_USER_PASSWORD') is not None:
                self.do_setting_activemq_groups("reads", os.getenv('ACTIVEMQ_READ_LOGIN') + "," + os.getenv(
                    'ACTIVEMQ_USER_LOGIN'))
            else:
                self.do_setting_activemq_groups("reads", os.getenv('ACTIVEMQ_READ_LOGIN'))

        # We setting the JMX access
        if os.getenv('ACTIVEMQ_JMX_LOGIN') is not None and os.getenv('ACTIVEMQ_JMX_PASSWORD') is not None:
            self.do_setting_activemq_jmx_access("readwrite", os.getenv('ACTIVEMQ_JMX_LOGIN'),
                                                      os.getenv('ACTIVEMQ_JMX_PASSWORD'))

        # We setting the log level
        if os.getenv('ACTIVEMQ_LOGLEVEL') is not None:
            self.do_setting_activemq_log4j(os.getenv('ACTIVEMQ_LOGLEVEL'))

        # We set the main parameters
        self.do_setting_activemq_main(os.getenv('ACTIVEMQ_NAME', 'localhost'),
                                            os.getenv('ACTIVEMQ_PENDING_MESSAGE_LIMIT', '1000'),
                                            os.getenv('ACTIVEMQ_STORAGE_USAGE', '100 gb'),
                                            os.getenv('ACTIVEMQ_TEMP_USAGE', '50 gb'),
                                            os.getenv('ACTIVEMQ_MAX_CONNECTION', '1000'),
                                            os.getenv('ACTIVEMQ_FRAME_SIZE', '104857600'),
                                            os.getenv('ACTIVEMQ_STATIC_TOPICS'), os.getenv('ACTIVEMQ_STATIC_QUEUES'),
                                            os.getenv('ACTIVEMQ_ENABLED_SCHEDULER', 'true'),
                                            os.getenv('ACTIVEMQ_ENABLED_AUTH', 'false'))

        # We setting wrapper
        self.do_setting_activemq_wrapper(os.getenv('ACTIVEMQ_MIN_MEMORY', '128'),
                                               os.getenv('ACTIVEMQ_MAX_MEMORY', '1024'))

if __name__ == '__main__':

    # We move all config file on temporary folder (Fix bug # 4)
    shutil.rmtree(ACTIVEMQ_CONF, ignore_errors=True);
    shutil.copytree(ACTIVEMQ_HOME + "/conf/", ACTIVEMQ_CONF);

    # We fix right on volume
    os.system("chown -R activemq:activemq /data/activemq")
    os.system("chown -R activemq:activemq " + ACTIVEMQ_CONF)
    os.system("chown -R activemq:activemq /var/log/activemq")

    serviceRun = Init()
    serviceRun.setting_all()
